### PR TITLE
Add an optional installer-host column to package lists

### LINKED ISSUE #5268: [IMPROVEMENT] Option to show installer host in detailed list of discovery & updates
### Please confirm these before moving forward.

- [x] I have searched for my feature proposal and have not found a work-in-progress/duplicate/resolved/discarded issue.
- [x] This improvement refers to an existing feature. If you want to suggest a new feature, please use [this template](https://github.com/Devolutions/UniGetUI/issues/new?labels=new-feature&projects=&template=feature-request.yml&title=%5BFEATURE+REQUEST%5D+%28Enter+your+description+here%29).
- [x] This improvement is not a bug. If you want to report a bug, please use [this template](https://github.com/Devolutions/UniGetUI/issues/new?labels=bug&projects=&template=bug-issue.yml&title=%5BBUG%5D+%28Enter+your+description+here%29).

### Describe the improvement

I just noticed a warning in my updates for a dotnet installer that the installer host had changed and be careful the URL has not been hijacked. This is great.

But it would be even better to know what the installers host are when searching in discovery or updating without having to right click into package details all the time. Perhaps an option to enable an additional column showing this for those whom wish to know where their packages are coming from?

### Describe how this improvement could help users

Empower users to be more conscious about where their packages are coming from. Increases security awareness. Users may also choose difference sources if the information is more forthcoming.

### PR DESCRIPTION
**Pull request by GabrielDuf**
This pull request introduces a new feature that displays the installer host (i.e., the domain from which a package installer will be downloaded) in package lists. The implementation includes UI, settings, caching, and backend logic to efficiently resolve, display, and optionally hide this information. Additionally, utility and test code has been added to robustly extract and format installer host details.

**Installer Host Column Feature:**

* Added a new optional "Installer host" column to package lists, including UI bindings, header text, and visibility logic controlled by a new user setting. 
* Introduced a new settings option "Show the installer host on package lists" in the interface settings, including localization and settings infrastructure. 

**Backend and Caching Logic:**

* Implemented logic in `PackageWrapper` to resolve, cache, and expose the installer host and its tooltip, including concurrency controls and a cache eviction strategy. 
* Added backend support for resolving installer URLs and hosts, including WinGet-specific logic and fallback mechanisms.

**Utility and Test Code:**

* Added `InstallerHostDisplay` utility for extracting and formatting hostnames from installer URLs, with robust handling of edge cases.
* Introduced comprehensive unit tests for installer host extraction and formatting logic.

**UI Infrastructure:**

* Added an attached property `PackageInstallerHostLoader.Track` to trigger installer host loading when a package row is rendered, ensuring on-demand and efficient data loading.

These changes collectively provide users with greater transparency about where package installers are sourced from, while maintaining performance and configurability.

### CODE REVIEW

File: src/UniGetUI.PackageEngine.Managers.WinGet/ClientHelpers/PingetPackageDetailsProvider.cs [hunk 1]
```diff
@@ -473,6 +473,35 @@ private static ShowResult ShowWithRepository(PackageQuery query)
...
+        return hosts.Count > 0 ? hosts : null;
+    }
+
+    internal static IReadOnlyList<string>? TryGetInstallerUrls(IPackage package, string? version)
+    {
+        IReadOnlyList<string>? urls = TryGetInstallerUrls(package, version, requireExactVersion: false);
+        if (urls is not null || string.IsNullOrWhiteSpace(version))
+            return urls;
+
+        return TryGetInstallerUrls(package, null, requireExactVersion: false);
```
[the remark this answers is not in the document]
Comment (GabrielDuf) [reply src/UniGetUI.PackageEngine.Managers.WinGet/ClientHelpers/PingetPackageDetailsProvider.cs]: [2f8234c](https://github.com/Devolutions/UniGetUI/pull/5320/commits/2f8234c0a85bdce0fe5653777205553fd68f5b0c)

File: src/UniGetUI.Avalonia/Models/PackageCollections.cs [hunk 2]
```diff
@@ -232,6 +241,111 @@ public Task EnsureIconLoadedAsync()
...
+            {
+                if (!TryGetCachedInstallerHost(hash, out resolved))
+                {
+                    IReadOnlyList<string>? urls = await ResolveInstallerUrlsAsync(token)
+                        .ConfigureAwait(false);
+                    resolved = (
+                        InstallerHostDisplay.FromUrls(urls),
+                        InstallerHostDisplay.JoinUrls(urls)
+                    );
+                    CacheInstallerHost(hash, resolved.Host, resolved.Urls);
```
[the remark this answers is not in the document]
Comment (GabrielDuf) [reply src/UniGetUI.Avalonia/Models/PackageCollections.cs]: [2f8234c](https://github.com/Devolutions/UniGetUI/pull/5320/commits/2f8234c0a85bdce0fe5653777205553fd68f5b0c)

File: src/UniGetUI.Avalonia/Models/PackageCollections.cs [hunk 3]
```diff
@@ -232,6 +241,111 @@ public Task EnsureIconLoadedAsync()
...
+#if WINDOWS
+        if (Package.Manager is WinGet)
+        {
+            string version = InstallerHostVersion;
+            return await Task.Run(() => WinGet.TryGetInstallerUrls(Package, version), token)
+                .ConfigureAwait(false);
+        }
+#endif
+        if (!Package.Details.IsPopulated)
+            await Package.Details.Load().ConfigureAwait(false);
```
[the remark this answers is not in the document]
Comment (GabrielDuf) [reply src/UniGetUI.Avalonia/Models/PackageCollections.cs:332]: [2f8234c](https://github.com/Devolutions/UniGetUI/pull/5320/commits/2f8234c0a85bdce0fe5653777205553fd68f5b0c)

File: src/UniGetUI.Avalonia/Models/PackageCollections.cs [hunk 4]
```diff
@@ -232,6 +244,157 @@ public Task EnsureIconLoadedAsync()
...
+                    IReadOnlyList<string>? urls = await ResolveInstallerUrlsAsync(token)
+                        .ConfigureAwait(false);
+                    resolved = (
+                        InstallerHostDisplay.FromUrls(urls),
+                        InstallerHostDisplay.JoinUrls(urls)
+                    );
+                    if (resolved.Host.Length > 0)
+                        CacheInstallerHost(hash, resolved.Host, resolved.Urls);
+                    else
+                        MarkInstallerHostUnresolved(hash);
```
[the remark this answers is not in the document]
Comment (GabrielDuf) [reply src/UniGetUI.Avalonia/Models/PackageCollections.cs:294]: [4539d48](https://github.com/Devolutions/UniGetUI/pull/5320/commits/4539d482ed61e014f439e267a7aa9d18e04e188d)
